android/filament-android/src/main/cpp/View.cpp

```diff
@@ -382,3 +384,29 @@ Java_com_google_android_filament_View_nIsScreenSpaceRefractionEnabled(JNIEnv *,
     View* view = (View*) nativeView;
     return (jboolean)view->isScreenSpaceRefractionEnabled();
 }
+
+extern "C"
+JNIEXPORT void JNICALL
+Java_com_google_android_filament_View_nPick(JNIEnv* env, jclass clazz,
+        jlong nativeView,
+        jint x, jint y, jobject handler, jobject internalCallback) {
+    View* view = (View*) nativeView;
+    JniCallback *callback = JniCallback::make(env, handler, internalCallback);
+    view->pick(x, y, [callback, env, internalCallback](View::PickingQueryResult const& result) {
+
+        auto internalOnPickCallbackClass = env->FindClass("com/google/android/filament/View$InternalOnPickCallback");
```

Comment:
Ideally you'd make those queries only once at init time. But For picking it's probably not an issue.

Reply:
Yes, I was not sure where to stash the result?

Reply:
A global structure is what we use in the platform. Usually we look them up in the JNI OnLoad but probably not worth it here.